site/source/hooks/useAxeCoreAnalysis.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useRef } from 'react'
+
+export const useAxeCoreAnalysis = () => {
+	// On stocke les packages importés dans des refs lors du premier import
+	// pour économiser les ressources
+	const axeRef = useRef<{
+		default: (_React: any, _ReactDOM: any, _timeout: number) => Promise<void>
+	}>()
+	const ReactRef = useRef<any>()
+	const ReactDOMRef = useRef<any>()
+
+	// useEffect déclenché à chaque re-render du composant ou des enfants du composant
+	// où il est placé
+	useEffect(() => {
+		// Ne se lance qu'en mode "development" avec le script "start:axe-debugging"
+		if (import.meta.env.PROD || !import.meta.env.VITE_AXE_CORE_ENABLED) return
+
+		const triggerAxeCoreAnalysis = async () => {
+			// On importe les packages requis uniquement si nécessaire
+			if (!axeRef.current && !ReactRef.current && !ReactDOMRef.current) {
+				axeRef.current = await import('@axe-core/react')
+				ReactRef.current = await import('react')
```

Comment:
C'est une bonne pratique d'importer uniquement axe-core si requis, mais je ne comprends pas pourquoi c'est fait à l'intérieur des hooks `useEffect`, et pourquoi on utilise une `ref`.

Dans leur doc, ils semblent dire que l'analyse est automatique, et ils mettent l'appel à la librairie en dehors des composants.

Tu as essayé de cette manière ? 
```js
import React from 'react'
import ReactDOM from 'react-dom'

if (import.meta.env.PROD || !import.meta.env.VITE_AXE_CORE_ENABLED)
  import('@axe-core/react').then(axe => axe.default(React, ReactDOM, 1000))
```
https://github.com/dequelabs/axe-core-npm/tree/develop/packages/react

Par ailleurs les bibliothèques `react` et `react-dom` sont déjà chargées : pas besoin de les importer via `await import`

Reply:
Alors j'ai essayé d'utiliser leur bout de code mais malheureusement il ne fonctionne pas comme attendu, l'analyse n'est effectuée que lors du premier rendering du composant ciblé et puis plus rien ne se passe quand on se promène dans l'app. Je ne sais pas si c'est idéal comme solution, mais des issues à ce sujet on été remontées et il ne semble pas que les auteurs de la lib soient très réactifs.. du coup si ça fonctionne et étant donné que c'est un outil de développement je me dis que ça peut faire l'affaire ! On peut jeter un oeil ensemble si besoin tu auras peut-être une idée que je n'ai pas eue

Reply:
Bien noté concernant les lib `React` et `ReactDOM` je fais la modif :)

Reply:
selon toi l'usage de la ref n'est pas nécessaire pour importer `axe-core/react` ?

Reply:
>  l'analyse n'est effectuée que lors du premier rendering du composant ciblé et puis plus rien ne se passe quand on se promène dans l'app. Je ne sais pas si c'est idéal comme solution, mais des issues à ce sujet on été remontées et il ne semble pas que les auteurs de la lib soient très réactifs

Ah mince, dommage. Pas de soucis pour importer axe-core dans une ref, mais tu peux le faire à l'extérieur du useEffect. Tu auras juste à checker si il est chargé où non dans useEffect avant de l'appeler. Sans oublier de l'appeler lorsque la promise est résolue également.

Reply:
Oui c'est bien ce que tu proposes ! C'est modifié, ça fonctionne comme une montre suisse ⌚